extensions/src/org/pentaho/platform/plugin/services/security/userrole/jdbc/JdbcUserRoleListService.java

```diff
@@ -146,7 +153,15 @@ public String getAllUsernamesQuery() {
       }
 
     }
-    return roles;
+
+    // Now add extra role if it does not exist in the list
+    for ( String extraRole : extraRoles ) {
+      if ( !roles.contains( extraRole ) ) {
```

Comment:
No need for this if statement if using a LinkedHashSet - that was the reason for the change. The LinkedHashSet will only allow unique values (like all other Set types).